modules/apps/frontend-icons/frontend-icons-web/src/main/java/com/liferay/frontend/icons/web/internal/servlet/taglib/SpritemapLocationTopHeadDynamicInclude.java

```diff
@@ -0,0 +1,68 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.frontend.icons.web.internal.servlet.taglib;
+
+import com.liferay.frontend.icons.util.FrontendIconsUtil;
+import com.liferay.portal.kernel.servlet.taglib.BaseDynamicInclude;
+import com.liferay.portal.kernel.servlet.taglib.DynamicInclude;
+import com.liferay.portal.kernel.util.ContentTypes;
+
+import java.io.IOException;
+import java.io.PrintWriter;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.osgi.service.component.annotations.Component;
+
+/**
+ * @author Bryce Osterhaus
+ */
+@Component(
+	immediate = true, property = "service.ranking:Integer=" + Integer.MAX_VALUE,
+	service = DynamicInclude.class
+)
+public class SpritemapLocationTopHeadDynamicInclude extends BaseDynamicInclude {
+
+	@Override
+	public void include(
+			HttpServletRequest httpServletRequest,
+			HttpServletResponse httpServletResponse, String key)
+		throws IOException {
+
+		PrintWriter printWriter = httpServletResponse.getWriter();
+
+		printWriter.write("<script data-senna-track=\"permanent\" type=\"");
+		printWriter.write(ContentTypes.TEXT_JAVASCRIPT);
+		printWriter.write("\">");
+		printWriter.write("window._FRONTEND_ICONS = {");
```

Comment:
I would put this under `Liferay` too...

Reply:
I put on window instead of Liferay for two reasons,

1. I didn't want to add more boilerplate to check if the Liferay object exists
2. I didn't want to further perpetuate the use of the Liferay object and add additional items to it.

This may not matter though if we decide to hardcode the values in the js util file, [see comment](https://github.com/liferay-frontend/liferay-portal/pull/2141#discussion_r857321224)

Reply:
Also, can we place things under window for avoiding ppl using window.Liferay.Icons._FRONTEND_ICONS and relying on this API. I mean, this can be a way to protect us and make it "private"